sql/core_clients_daily_v1.sql

```diff
@@ -0,0 +1,54 @@
+WITH
+  windowed AS (
+  SELECT
+    @submission_date AS submission_date,
+    CURRENT_DATETIME() AS generated_time,
+    client_id,
+    ROW_NUMBER() OVER w1 AS n,
+    -- For now, we're ignoring the following RECORD type fields:
+    --   accessibility_services
+    --   experiments
+    --   searches
+    --
+    -- Take the min observed profile creation date.
+    DATE_FROM_UNIX_DATE(MIN(profile_date) OVER w1) AS profile_date,
+    -- These integer fields are already sums over sessions since last upload,
+    -- so we sum to represent all uploads in the given day.
+    SUM(durations) OVER w1 AS durations,
+    SUM(flash_usage) OVER w1 AS flash_usage,
+    -- For all other fields, we take the value in the most recently received ping.
+    LAST_VALUE(app_name) OVER w1 AS app_name,
```

Comment:
current `clients_daily` generally uses `first_value` for this sort of thing. cc @mreid-moz for thoughts

Reply:
Thoughts on instead using `MODE`? I'm happy to see that this is idempotent with the `ORDER BY` clause but I'd imagine `MODE` is more useful (we could use `APPROX_TOP_COUNT` for this).

The rationale is that `mode` is probably more uniform for a client. If they have indeed changed one of their values at some point during the day (such that `last` and `mode` are different), then tomorrow the `mode` will probably change as well.

Reply:
Desirable qualities:
- stable when re-computed
- interpretable
- not systematically biased (like `MIN` or `MAX` would be)
- resilient to null values
- simple

Given these, my ideal would be a `MODE`-like function that skips nulls and has a predictable tie-breaker which is either "first seen" or "last seen" (I don't feel super strongly re: first v. last). That only conflicts with the "simple" quality :)

Reply:
Unfortunately, if we use `APPROX_TOP_COUNT` it will not be idempotent, since we can't guarantee a stable result every time.

Reply:
Using LAST_VALUE is easier to explain and reason about. In particular, all the values we get will be from the same ping, where doing a MODE could mean mixing dimension values from different pings.

The next step here is going to be a `core_clients_last_seen` dataset in analogy to desktop `clients_last_seen` and data science folks just okay'ed yesterday the methodology of using the most recent observation for all dimension values. I like the idea of keeping identical methodology for the daily view and the last_seen view such that rows in last_seen are truly showing you dimension values based on the most recently received ping up to the given date.

For applications other than last_seen, though, I sympathize that MODE does indeed seem more correct. Perhaps we can capture that as an issue to investigate for the future and measure what effect it would have to use that method? My gut is telling me to stick with the simpler and more easily explained methodology absent analysis that suggests the more complex solution is worth it.

Reply:
> resilient to null values

That's a good point. If we stick with LAST_VALUE, we can add IGNORE NULLS. That abandons the idea that the dimensions all consistently describe a single ping, though perhaps that's not really an important design goal.

> Unfortunately, if we use APPROX_TOP_COUNT it will not be idempotent, since we can't guarantee a stable result every time.

I'd agree that we should avoid APPROX_TOP_COUNT for lack of idempotence. I haven't found an alternative, though, that doesn't involve doing fairly complicated array manipulation and potentially a js UDF.

Reply:
> That's a good point. If we stick with LAST_VALUE, we can add IGNORE NULLS. That abandons the idea that the dimensions all consistently describe a single ping, though perhaps that's not really an important design goal.

At this point it looks like there's no great reason to use `LAST_VALUE` anymore except that it's easier to compute. Keeping tack with desktop shouldn't be a requirement if we indeed consider `mode` a better aggregator for these values. Eventually we can move desktop to a different aggregation as well, see bug 1515670. What are your thoughts @jklukas?

Reply:
> data science folks just okay'ed yesterday the methodology of using the most recent observation for all dimension values.

🎉 

> For applications other than last_seen, though, I sympathize that MODE does indeed seem more correct. Perhaps we can capture that as an issue to investigate for the future and measure what effect it would have to use that method? My gut is telling me to stick with the simpler and more easily explained methodology absent analysis that suggests the more complex solution is worth it.

There's no reason we couldn't have multiple aggregations of the same field, so we could theoretically store both "last" and "mode" as separate columns if need be.

Reply:
ah, i thought `IGNORE NULLS` was the default. we should either add that or reverse the sort order for `ROW_NUMBER` and not window the `LAST_VALUE` fields for the same effect.

Reply:
I had declared `mode` impossible to implement efficiently, but @fbertsch gave me a push and it appears to be totally doable. I've pushed a new commit that implement `udf_mode_last` that takes in an array of values, picks the most frequently occurring value, and breaks ties by using the last value observed.

Reply:
There is no discernible performance different between using this UDF and array agg compared to using LAST_VALUE. Either way, it takes ~70 seconds to process a single day.